apps/api/src/domain/models/audio-item/commands/transcripts/translate-line-item/translate-line-item.command.integration.spec.ts

```diff
@@ -0,0 +1,180 @@
+            commandHandlerService,
+            idManager,
+        };
+    });
+
+    afterAll(async () => {
+        await app.close();
+    });
+
+    beforeEach(async () => {
+        await testRepositoryProvider.testSetup();
+    });
+
+    afterEach(async () => {
+        await testRepositoryProvider.testTeardown();
+    });
+
+    describe(`when the command is valid`, () => {
+        it(`should succeed with the expected database updates`, async () => {
+            await assertCommandSuccess(assertionHelperDependencies, {
+                systemUserId: dummySystemUserId,
+                initialState: new DeluxeInMemoryStore({
+                    [existingAudioItem.type]: [existingAudioItem],
+                }).fetchFullSnapshotInLegacyFormat(),
+                buildValidCommandFSA: () => buildValidFsa(existingAudioItem),
+            });
+        });
+    });
+
+    describe(`when the command is invalid`, () => {
+        describe(`when the audio-visual resource does not exist`, () => {
+            it(`should fail with the expected error`, async () => {
+                await assertCommandError(assertionHelperDependencies, {
+                    systemUserId: dummySystemUserId,
+                    initialState: new DeluxeInMemoryStore({}).fetchFullSnapshotInLegacyFormat(),
+                    buildCommandFSA: () => buildValidFsa(existingAudioItem),
+                });
+            });
+        });
+
+        describe(`when the line item already has text in this language`, () => {
+            it(`should fail with the expected errors`, async () => {
+                await assertCommandError(assertionHelperDependencies, {
+                    systemUserId: dummySystemUserId,
+                    initialState: new DeluxeInMemoryStore({
+                        [existingAudioItem.type]: [
+                            existingAudioItem.clone({
+                                transcript: existingAudioItem.transcript.clone({
+                                    items: [
+                                        new TranscriptItem({
+                                            inPointMilliseconds: 0,
+                                            outPointMilliseconds: 0.001,
+                                            speakerInitials: targetSpeakerInitials,
+                                            text: buildMultilingualTextWithSingleItem(
+                                                'I already have text in this language',
+                                                translationLanguageCode
+                                            ),
+                                        }),
+                                    ],
+                                }),
+                            }),
+                        ],
+                    }).fetchFullSnapshotInLegacyFormat(),
+                    buildCommandFSA: () => buildValidFsa(existingAudioItem),
+                });
+            });
+        });
+    });
```

Comment:
Missing test coverage:
- when the inPointMilliseconds does not match an existing line item
- when the outPointMilliseconds does not match an existing line item

Reply:
This has since been done.